go-controller/pkg/cni/cni_dpu.go

```diff
@@ -4,9 +4,35 @@ import (
 	"fmt"
 
 	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/kube"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/types"
 	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+
+	"k8s.io/client-go/util/retry"
 )
 
+// updatePodDPUConnDetailsWithRetry update the pod annotion with the givin connection details
+func (pr *PodRequest) updatePodDPUConnDetailsWithRetry(kube kube.Interface, dpuConnDetails *util.DPUConnectionDetails) error {
+	resultErr := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+		// Informer cache should not be mutated, so get a copy of the object
+		pod, err := kube.GetPod(pr.PodNamespace, pr.PodName)
+		if err != nil {
+			return err
+		}
+
+		cpod := pod.DeepCopy()
+		err = util.MarshalPodDPUConnDetails(&cpod.Annotations, dpuConnDetails, types.DefaultNetworkName)
+		if err != nil {
+			return err
+		}
+		return kube.UpdatePod(cpod)
+	})
+	if resultErr != nil {
+		return fmt.Errorf("failed to update %s annotation on pod %s/%s: %v",
+			util.DPUConnectionDetailsAnnot, pr.PodNamespace, pr.PodName, resultErr)
+	}
+	return nil
+}
```

Comment:
We should move the dpu handlers into obj_retry and then we would not need this retry nor fetching the pod from the api server